docs/DeveloperGuide.md

```diff
@@ -1,38 +1,536 @@
+- `shuffle` 
+
+#### Class Structure of Shuffle Command
+![commandShuffleSequence.png](UML%2FCommands%2FcommandShuffleSequence.png)
+
+The "shuffle" command in QuizHub is used to shuffle the order of questions within a question list. 
+This command provides users with the ability to randomize the sequence of questions, which can be useful for 
+creating randomized quizzes or study sessions. 
+
+![ShuffleToStorage-Shuffle_to_Storage_Flow.png](UML%2FCommands%2FShuffleToStorage-Shuffle_to_Storage_Flow.png)
+
+**NOTE:** The randomised sequence will be stored PERMANENTLY in storage. This is unlike /random for command start which 
+creates a temporary array to store the randomised sequence of questions
+
+
+### Markdiff Command - mark difficulty of entry
+
+#### Brief Description of Markdiff Command
+The CommandMarkDifficulty class in the quizhub application is responsible for handling user commands to mark the 
+difficulty of a question. 
+
+The CommandMarkDifficulty class supports the following command syntax:
+- `markdiff [question number] /[question difficulty]` - sets the difficulty of question
+
+#### Class Structure of Markdiff Command
+![commandMarkDiffSequence.png](UML%2FCommands%2FcommandMarkDiffSequence.png)
+
+![commandMarkDiffClass.png](UML%2FCommands%2FcommandMarkDiffClass.png)
+The CommandMarkDifficulty class includes the following key components:
+- `qnIndex`: An integer representing the question number to be marked for difficulty.
+- `nDifficulty`: An enumeration representing the difficulty level to be assigned to the question.
+
+#### Implementation of Markdiff Command
+Developers can use the `CommandMarkDifficulty` class as a template for handling difficulty marking commands in the 
+quizhub application. Here are the key steps for implementing this class:
+
+- **Parsing User Input**: Parse the user input to extract the question number and the specified difficulty level.
+
+- **Validation**: Implement validation logic to ensure that the user input is correctly formatted and contains valid 
+information.
+
+- **Marking Difficulty**: Implement the logic to execute the operation of marking the question with the specified 
+difficulty level. This typically involves invoking methods in the QuestionList and Question classes to update the 
+uestion's difficulty.
+
+- **Data Persistence**: If necessary, update the data storage to save the changes. In the provided code, the 
+`dataStorage.updateData(questions)` method is used to save changes to the question list.
+
+- **Error Handling**: Handle any exceptions or errors that may occur during the marking operation and provide 
+appropriate feedback to the user.
+
+
+### Command Exit - Exit Program
+
+#### Brief Description of Command Exit
+
+Command Exit is responsible for exiting the program
+- `bye` - to exit program
+
+
+#### Class structure of Command Exit
+
+![commandExitSequence.png](UML%2FCommands%2FcommandExitSequence.png)
```

Comment:
Should the return arrow be a dotted line?

![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/94791461/db165ff3-6b78-4624-a083-5461bd82a295)